src/twisted/logger/_format.py

```diff
@@ -34,27 +34,12 @@ def formatEvent(event):
     @return: A formatted string.
     @rtype: L{unicode}
     """
-    try:
-        if "log_flattened" in event:
-            return flatFormat(event)
-
-        format = event.get("log_format", None)
-        if format is None:
-            return u""
-
-        # Make sure format is unicode.
-        if isinstance(format, bytes):
-            # If we get bytes, assume it's UTF-8 bytes
-            format = format.decode("utf-8")
-        elif not isinstance(format, unicode):
-            raise TypeError(
-                "Log format must be unicode or bytes, not {0!r}".format(format)
-            )
-
-        return formatWithCall(format, event)
-
-    except Exception as e:
-        return formatUnformattableEvent(event, e)
+    return eventAsText(
+        event,
+        includeTraceback=False,
+        includeTimestamp=False,
+        includeSystem=False,
+    )[:-1]
```

Comment:
This `:-1` seems like a rather unfortunate consequence of the newline being included.  I realize that this is a "legacy" API now, ideally destined to be deprecated... but still, giving it intimate knowledge of the internals of another function seems weird.  (Why `:-1`?  What about `\r\n` newlines?  What about record separators? etc.)